docs/public/components/creating-components.md

```diff
@@ -90,23 +89,62 @@ For our compression example is will look like this:
     }
     ```
 
-All component interfaces must be called `Component` so all imports would be similar to `compression.Component`.
+All component interfaces must be called `Component` so all imports would be similar to `compressiondef.Component`.
 
 You can see that our interface only exposes the bare minimum. You should aim at having the smallest possible interface
-for your component. Also note that there is no `Start`/`Stop` method. Anything related to lifecycle will be handle
+for your component. 
+
+When defining a component interface we try to use struct or interface from third-party dependencies. Ensuring that using the component interface doesn't require importing any thrid-party dependency.
+
+!!! warning "Interface using a third-party dependency"
+    ```
+    package telemetrydef
+
+    import "github.com/prometheus/client_golang/prometheus"
+
+    // team: agent-shared-components
+
+    // Component is the component type.
+    type Component interface {
+        // RegisterCollector Registers a Collector with the prometheus registry
+        RegisterCollector(c prometheus.Collector)
+    }
+    ```
+
+For the example above every user of the telemetry component would have to import `github.com/prometheus/client_golang/prometheus`. To avoid those cases is best to design your interface mindfully.
+
+TODO: write alternative solution
```

Comment:
```suggestion
For the example above every user of the `telemetry` component would have to import `github.com/prometheus/client_golang/prometheus` no matter which implementation they use.

In general, be mindful of using external types in the public interface of your component. For example, it would make sens to use docker types in a `docker` component but not in a `container` component.
```